examples/llama.py

```diff
@@ -10,48 +10,45 @@
 # model=og.Model("../../test_models/llama2-7b-fp32-cpu", device_type)
 #model=og.Llama_Model("../../test_models/llama2-7b-fp16-gpu/rank_0_Llama-2-7b-hf_decoder_merged_model_fp16.onnx", device_type)
 #model=og.Llama_Model("../../test_models/llama2-7b-int4-gpu/rank_0_Llama-2-7b-hf_decoder_merged_model_int4.onnx", device_type)
-model=og.Model("../../test_models/llama2-7b-chat-int4-gpu", device_type)
+model=og.Model("../test_models/llama2-7b-chat-int4-gpu", device_type)
 print("Model loaded")
-# tokenizer = LlamaTokenizer.from_pretrained('meta-llama/Llama-2-7b-hf')
-tokenizer=model.CreateTokenizer()
+tokenizer = LlamaTokenizer.from_pretrained('meta-llama/Llama-2-7b-hf')
+# tokenizer=model.CreateTokenizer()
 print("Tokenizer created")
 
 # Keep asking for input prompts in an loop
 while True:
     text = input("Input:")
-    # input_tokens = tokenizer.encode(text, return_tensors='np')
-    input_tokens = tokenizer.encode(text)
+    input_tokens = tokenizer.encode(text, return_tensors='np')
+    # input_tokens = tokenizer.encode(text)
 
     params=og.SearchParams(model)
     params.max_length = 128
     params.input_ids = input_tokens
 
-    search=params.CreateSearch()
-    state=model.CreateState(search.GetSequenceLengths(), params)
+    generator=og.Generator(model, params)
 
     print("Output:")
 
     print(text, end='', flush=True)
-    while not search.IsDone():
-        search.SetLogits(state.Run(search.GetSequenceLength(), search.GetNextTokens()))
+    while not generator.IsDone():
+        generator.ComputeLogits()
 
         # search.Apply_MinLength(1)
         # search.Apply_RepetitionPenalty(1.0)
 
-        search.SampleTopP(0.7, 0.6)
+        generator.AppendNextToken_TopP(0.7, 0.6)
```

Comment:
Append doesn't look like a good name to me. generator actually generate next token.

Reply:
And can we make TopP a parameter instead of part of the name?

Reply:
I named it that since you first compute the logits, then append the next token based on the logits.
I initially had AddNextToken but 'Add' was less correct than 'Append' since it's appending tokens, not adding them.
Did you have a better name?

Reply:
It is a process of generate next token. Add or Append doesn't reflect the action, i think.